### PR TITLE
cmd/tailscaled,*: add start of configuration file support

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #1412

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -518,6 +521,25 @@ func (b *LocalBackend) SetDirectFileDoFinalRename(v bool) {
...
+	if b.conf == nil {
+		return false, nil
+	}
+	conf, err := conffile.Load(b.conf.Path)
+	if err != nil {
+		return false, err
+	}
+	b.conf = conf
+	// TODO(bradfitz): apply things
+	return true, nil
```
Comment (maisem) [inline ipn/ipnlocal/local.go:540]: return false, nil for now?

Comment (maisem) [reply ipn/ipnlocal/local.go:540]: or false, errors.New("unimplemented")

**Review by maisem — APPROVED** [review]

File: ipn/conf.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+// tailscale set --hostname=foo
+// WARNING: DIRTY CONFIG
+// tailscale  status
+// WARNING DIRTY DIRTY
+
+// tailscale set --reload-config   # revert changes to what's on disk
+// tailscale write-config          # write to disk from current state
+// TODO: command to write imperative config to this new style JSON format to make it declarative
+
+type ConfigVersionOnly struct {
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply ipn/conf.go:27]: Deleted. Moved to the parser code.

I don't feel the need to add version to the VAlpha struct. I don't want callers doing anything with it. And is it the pre- or post-upgraded version? Not worth documenting and no need if we just remove it.

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (bradfitz) [reply ipn/ipnlocal/local.go:540]: No, that's how I was testing the CLI/API changes.

**Pull request merged**